packages/turf-points-within-polygon/index.d.ts

```diff
@@ -14,6 +15,8 @@ export default function pointsWithinPolygon<
   G extends Polygon | MultiPolygon,
   P = Properties
 >(
-  points: Feature<Point, P> | FeatureCollection<Point, P>,
+  points:
+    | Feature<Point | MultiPoint, P>
+    | FeatureCollection<Point | MultiPoint, P>,
   polygons: Feature<G> | FeatureCollection<G> | G
-): FeatureCollection<Point, P>;
+): FeatureCollection<Point | MultiPoint, P>;
```

Comment:
The downside to this return signature (in Typescript anyway) is that it narrowly implies the FC can contain Points or MultiPoints but in fact it is one or the other, not both.  So the user can run into some friction that requires type narrowing.  This may be acceptable because the two types are so similar, varying only in number of coordinate Positions.  Otherwise, a solution to overcome is Typescript function overloading